pkg/controller/jobs/deployment/deployment_webhook.go

```diff
@@ -100,12 +98,12 @@ func (wh *Webhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Ob
 	oldQueueName := jobframework.QueueNameForObject(oldDeployment.Object())
 	newQueueName := jobframework.QueueNameForObject(newDeployment.Object())
 
-	allErrs := apivalidation.ValidateImmutableField(oldQueueName, newQueueName, queueNameLabelPath)
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(
-		newDeployment.Spec.Template.GetLabels()[constants.QueueLabel],
-		oldDeployment.Spec.Template.GetLabels()[constants.QueueLabel],
-		podSpecQueueNameLabelPath,
-	)...)
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, jobframework.ValidateQueueName(newDeployment.Object())...)
```

Comment:
Add assert to check that queue-name cannot be removed, because this remains broken in the PR, and add unit test to check that.

Reply:
Done

Reply:
which test case shows that the queue-name cannot be removed?

Reply:
https://github.com/kubernetes-sigs/kueue/pull/3528/files#diff-afdbc155b42eaaba565176e6b31c0611aecb1c8b8989f13d4914be3cc24bf545R51-R64

Reply:
Sorry, I meant that we should still block dropping the queue-name on the Deployment. So oldQueueName!= nil, newQueueName=nil, should be blocked.

Reply:
Done